control_data_collecting_tool/scripts/data_collecting_pure_pursuit_trajectory_follower_actuation_cmd.py

```diff
@@ -0,0 +1,675 @@
+                * self.get_parameter("steer_noise_amp").get_parameter_value().double_value
+            )
+            noise_min_period = (
+                self.get_parameter("steer_noise_min_period").get_parameter_value().double_value
+            )
+            noise_max_period = (
+                self.get_parameter("steer_noise_max_period").get_parameter_value().double_value
+            )
+            tmp_noise_period = noise_min_period + np.random.rand() * (
+                noise_max_period - noise_min_period
+            )
+
+            dt = self.timer_period_callback
+            noise_data_num = max(4, int(tmp_noise_period / dt))
+            for i in range(noise_data_num):
+                self.steer_noise_list.append(
+                    tmp_noise_amp * np.sin(2.0 * np.pi * i / noise_data_num)
+                )
+
+        # [1] compute control
+        targetIndex = 1 * nearestIndex
+        lookahead_coef = self.get_parameter("lookahead_time").get_parameter_value().double_value
+        lookahead_intercept = self.get_parameter("min_lookahead").get_parameter_value().double_value
+        pure_pursuit_lookahead_length = (
+            lookahead_coef * present_linear_velocity[0] + lookahead_intercept
+        )
+
+        while True:
+            tmp_distance = np.sqrt(
+                ((trajectory_position[targetIndex][:2] - present_position[:2]) ** 2).sum()
+            )
+            if tmp_distance > pure_pursuit_lookahead_length:
+                break
+            if targetIndex == (len(trajectory_position) - 1):
+                break
+            targetIndex += 1
+
+        pure_pursuit_type = (
+            self.get_parameter("pure_pursuit_type").get_parameter_value().string_value
+        )
+
+        steer_cmd_without_limit = 0.0
+        if pure_pursuit_type == "naive":
+            steer_cmd_without_limit = self.pure_pursuit_steer_control(
+                present_position[:2],
+                present_yaw,
+                present_linear_velocity[0],
+                trajectory_position[targetIndex][:2],
+            )
+        elif pure_pursuit_type == "linearized":
+            steer_cmd_without_limit = self.linearized_pure_pursuit_steer_control(
+                present_position[:2],
+                present_yaw,
+                present_linear_velocity[0],
+                trajectory_position[targetIndex][:2],
+                getYaw(trajectory_orientation[targetIndex]),
+            )
+        else:
+            self.get_logger().info(
+                'pure_pursuit_type should be "naive" or "linearized" but is set to "%s" '
+                % pure_pursuit_type
+            )
+
+        acceleration_cmd = self.get_parameter("stop_acc").get_parameter_value().double_value
+        if self.CONTROL_MODE == "actuation_cmd" or self.CONTROL_MODE == "external_actuation_cmd":
+            if self.pedal_input is not None:
+                acceleration_cmd = self.accel_brake_map.convert_actuation_cmd_to_accel_input(
```

Comment:
memo:

actuation_cmd is not published directly, reconverted in raw_vehicle_cmd_converter.
actuation_cmd -[this tool]-> acceleration_cmd -[raw_vehicle_cmd_converter]-> actuation_cmd